rs-commerce/src/views/user-profile/button-profile/ButtonProfile.tsx

```diff
@@ -0,0 +1,36 @@
+import { useAppSelector } from 'hooks/typed-react-redux-hooks';
+import { updateProfileSelector } from 'redux/selectors';
+import classes from './buttonProfile.module.css';
+
+export type BtnProfilePropsType = {
+  content: string;
+  colored: boolean;
+  onClick: () => void;
+  page?: 'password' | 'name';
+};
+
+export default function ButtonProfile(props: BtnProfilePropsType): JSX.Element {
+  const { content, colored, onClick, page } = props;
+
+  const { checkNewPassword, checkNewName } = useAppSelector(updateProfileSelector);
+
+  return (
+    <button
+      type="button"
+      className={
+        colored && (checkNewPassword || checkNewName)
+          ? `${classes['profile__button']} ${classes['profile__button_colored']}`
+          : `${classes['profile__button']}`
+      }
+      onClick={onClick}
+      disabled={
+        page === 'password'
+          ? !!(content === 'Save' && !checkNewPassword)
+          : page === 'name'
+            ? !!(content === 'Save' && !checkNewName)
+            : false
```

Comment:
use `const` variable for content to avoid spelling errors and then errors in `disabled` functionality